scripts/tools/silabs/FactoryDataProvider.py

```diff
@@ -0,0 +1,212 @@
+import subprocess
+import sys
+import argparse
+import struct
+
+# global variables
+Discriminator_ui = 0
+Passcode_ui = 0
+IterationCount_ui = 0
+Salt_str = ""
+Verifier_str = ""
+mcuFamily = ""
+jtagSerial = ""
+isDeviceConnected = True
+
+# CONSTANTS
+TEMP_FILE = 'tmp_nvm3.s37'
+OUT_FILE = 'matter_factorydata.s37'  # Final output file containing the nvm3 data to flash to the device
+# nvm3 keys to set
+PASSCODE_NVM3_KEY = "0x87205:"
+DISCRIMINATOR_NVM3_KEY = "0x87207:"
+ITERATIONCOUNT_NVM3_KEY = "0x87208:"
+SALT_NVM3_KEY = "0x87209:"
+VERIFIER_NVM3_KEY = "0x8720A:"
+
+
+def generate_spake2p_verifier(args):
```

Comment:
You can see how Espressif did it with `spake2p.py` rather than relying on the native `spake2p` generator.

Reply:
I could not find such example or any information about spake2p.py

However, in script/tools both nordic and espressif use the native spake2p generator in the factory data generator tools as I am using here.

I'll leave it like this for this iteration of the tool